src/App.tsx

```diff
@@ -8,35 +8,131 @@ import { PostsList } from './components/PostsList';
+    let timer: ReturnType<typeof setTimeout>;
+
+    if (error) {
+      timer = setTimeout(() => {
+        setError(false);
+      }, 4000);
+    }
+
+    return () => clearTimeout(timer);
+  }, [error]);
+
+  const handleUserSelect = (user: User) => {
+    setSelectedUser(user);
+    setListOfUserPosts([]);
+    setSelectedPost(null);
+
+    if (noPosts) {
+      setNoPosts(false);
+    }
+
+    setLoadingListOfPosts(true);
+
+    getAllUserPosts(user.id)
+      .then(response => {
+        setListOfUserPosts(response);
+        setLoadingListOfPosts(false);
+
+        if (!response.length) {
+          setNoPosts(true);
+        }
+      })
+      .catch(() => {
+        setError(true);
+        setLoadingListOfPosts(false);
+      });
+  };
+
+  const handlePostSelect = async (id: number) => {
+    if (id === -1) {
+      setSelectedPost(null);
+
+      return;
+    }
+
+    setLoadingPost(true);
+
+    if (listOfUserPosts) {
+      const findPost = listOfUserPosts.find(post => post.id === id);
+
+      if (findPost) {
+        setSelectedPost(findPost);
+
+        getCommentsByPostId(id)
+          .then(res => {
+            setListOfComments(res);
+          })
+          .catch(() => setErrorComment(true))
+          .finally(() => {
+            setLoadingPost(false);
+          });
+      }
+    } else {
+      setErrorComment(true);
+    }
+  };
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  onSelectUser={handleUserSelect}
+                  selectedUser={selectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <Loader />
+                {loadingListOfPosts && <Loader />}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {error && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {noPosts && (
+                  <div className="notification is-warning" data-cy="NoPostsYet">
+                    No posts yet
+                  </div>
+                )}
 
```

Comment:
You have `UserSelector` responsible for loading users, so it makes sense make `PostsList` responsible for loading posts, displaying error and loader sections, etc

Reply:
I had difficulty with tests.
And there is a need to send a bunch of props inside PostList in case we transfer the logic(